File: app/src/main/java/by/kirich1409/grsuschedule/widget/FreeTimeViewHolder.java
package by.kirich1409.grsuschedule.widget;

import android.content.Context;
import android.view.View;

import org.jetbrains.annotations.NotNull;

import by.kirich1409.grsuschedule.app.BaseRecyclerItemViewHolder;
import by.kirich1409.grsuschedule.databinding.ItemFreeTimeBinding;
import by.kirich1409.grsuschedule.schedule.FreeTime;

/**
 * Created by kirillrozov on 10/10/15.
 */
public class FreeTimeViewHolder extends BaseRecyclerItemViewHolder {

    final ItemFreeTimeBinding mBinding;

    public FreeTimeViewHolder(@NotNull Context context, @NotNull View rootView) {
        super(context, rootView);
        mBinding = ItemFreeTimeBinding.bind(rootView);
    }

    public void setFreeTime(FreeTime freeTime) {
        mBinding.setFreeTime(freeTime);
        mBinding.executePendingBindings();
    }
}

File: app/src/main/kotlin/by/kirich1409/grsuschedule/widget/FreeTimeViewHolder.kt
package by.kirich1409.grsuschedule.widget

import android.content.Context
import android.view.View
import android.widget.TextView
import by.kirich1409.grsuschedule.R
import by.kirich1409.grsuschedule.app.BaseRecyclerItemViewHolder
import by.kirich1409.grsuschedule.model.FreeTime
import by.kirich1409.grsuschedule.utils.bindView

/**
 * Created by kirillrozov on 9/25/15.
 */
public class FreeTimeViewHolder(context: Context, rootView: View) :
        BaseRecyclerItemViewHolder(context, rootView) {

    val textView: TextView by bindView(android.R.id.text1)
    var freeTime: FreeTime? = null
        set(freeTime) {
            field = freeTime
            val interval = freeTime?.interval
            if (interval != null) {
                textView.text = context.getString(
                        R.string.free_time_format, interval.startTime, interval.endTime)
            } else {
                textView.text = null
            }
        }
}